src/mpid/ch4/src/ch4_comm.h

```diff
@@ -201,6 +203,10 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_Comm_create_hook(MPIR_Comm * comm)
         }
     }
 
+#if defined HAVE_LIBHCOLL
+    hcoll_comm_create(comm, NULL);
+#endif
+
```

Comment:
In which case, I am assuming we can safely move this `hcoll_comm_create` down into the UCX NM hook.